opensrp-chw/src/miecd/java/org/smartregister/chw/interactor/PncHomeVisitInteractorFlv.java

```diff
@@ -0,0 +1,953 @@
+                .build();
+        actionList.put(context.getString(R.string.pnc_danger_signs_mother), action);
+    }
+
+    private void evaluateDangerSignsBaby() throws Exception {
+        class PNCDangerSignsBabyHelper extends HomeVisitActionHelper {
+            private String danger_signs_present_child;
+
+            @Override
+            public void onPayloadReceived(String s) {
+                try {
+                    JSONObject jsonObject = new JSONObject(s);
+                    danger_signs_present_child = org.smartregister.chw.util.JsonFormUtils.getCheckBoxValue(jsonObject, "danger_signs_present_child");
+                } catch (JSONException e) {
+                    Timber.e(e);
+                }
+            }
+
+            @Override
+            public String evaluateSubTitle() {
+                return MessageFormat.format("{0}: {1}", context.getString(R.string.pnc_danger_signs_baby_task), danger_signs_present_child);
+            }
+
+            @Override
+            public BaseAncHomeVisitAction.Status evaluateStatusOnPayload() {
+                if (StringUtils.isBlank(danger_signs_present_child)) {
+                    return BaseAncHomeVisitAction.Status.PENDING;
+                }
+
+                if (StringUtils.isNotBlank(danger_signs_present_child)) {
+                    return BaseAncHomeVisitAction.Status.COMPLETED;
+                } else {
+                    return BaseAncHomeVisitAction.Status.PARTIALLY_COMPLETED;
+                }
+            }
+        }
+
+        for (Person baby : children) {
+            if (getAgeInDays(baby.getDob()) <= 28) {
+                Map<String, List<VisitDetail>> details = null;
+                if (editMode) {
+                    Visit lastVisit = AncLibrary.getInstance().visitRepository().getLatestVisit(baby.getBaseEntityID(), Constants.EventType.DANGER_SIGNS_BABY);
+                    if (lastVisit != null) {
+                        details = VisitUtils.getVisitGroups(AncLibrary.getInstance().visitDetailsRepository().getVisits(lastVisit.getVisitId()));
+                    }
+                }
+
+                BaseAncHomeVisitAction action = new BaseAncHomeVisitAction.Builder(context, MessageFormat.format(context.getString(R.string.pnc_danger_signs_baby), baby.getFullName()))
+                        .withOptional(false)
+                        .withDetails(details)
+                        .withBaseEntityID(baby.getBaseEntityID())
+                        .withProcessingMode(BaseAncHomeVisitAction.ProcessingMode.SEPARATE)
+                        .withFormName(Constants.JSON_FORM.PNC_HOME_VISIT.getDangerSignsBaby())
+                        .withHelper(new PNCDangerSignsBabyHelper())
+                        .build();
+                actionList.put(MessageFormat.format(context.getString(R.string.pnc_danger_signs_baby), baby.getFullName()), action);
+            }
+        }
+    }
+
+    private void evaluateFamilyPlanning() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'evaluateFamilyPlanning()' has an NPath complexity of 16128](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5426631)